sdk/storage/storage-file-datalake/src/utils/tracing.ts

```diff
@@ -21,12 +21,17 @@ export function createSpan(
   };
 
   const span = tracer.startSpan(`Azure.Storage.DataLake.${operationName}`, spanOptions);
+  span.setAttribute("az.namespace", "Microsoft.Storage");
 
   let newOptions = tracingOptions.spanOptions || {};
   if (span.isRecording()) {
     newOptions = {
-      ...tracingOptions,
-      parent: span
+      ...spanOptions,
+      parent: span,
+      attributes: {
+        ...spanOptions.attributes,
+        "az.namespace": "Microsoft.Storage"
+      }
```

Comment:
just curious: why az.namespace need to be set at two places, to span, then in the options?

Reply:
The first one sets it on the span created in the convenience layer, the second one updates options so that the core-http span will also have it set. 

There's no way to interrogate a Span for its attributes, so core-http couldn't just look into what the parent span had in its attribute bag.